README.md

```diff
@@ -14,8 +14,8 @@ Naturally, CI/CD events contain information that should:
 
 The Tekton API enables functionality to be seperated from configuration (e.g. [Pipelines](https://github.com/tektoncd/pipeline/blob/master/docs/pipelines.md) vs [PipelineRuns](https://github.com/tektoncd/pipeline/blob/master/docs/pipelineruns.md)) such that steps can be reusable, but it does not provide a mechanism to generate the resources (notably, [PipelineRuns](https://github.com/tektoncd/pipeline/blob/master/docs/pipelineruns.md) and [PipelineResources](https://github.com/tektoncd/pipeline/blob/master/docs/resources.md#pipelineresources)) that encapsulate these configurations dynamically. Triggers extends the Tekton architecture with the following CRDs:
 - [`TriggerTemplate`](docs/triggertemplates.md) - Templates resources to be created (e.g. Create PipelineResources and PipelineRun that uses them)
-- [`TriggerBinding`](docs/triggerbindings.md) - Instantiates resources in TriggerTemplate using event fields
-- [`EventListener`](docs/eventlisteners.md) - Wraps TriggerBinding(s) into an addressable endpoint (the event sink)
+- [`TriggerBinding`](docs/triggerbindings.md) - Validates events and extracts payload fields
```

Comment:
I feel like the fact that we are describing the functionality of this with an "and" is a strong clue that this object now has two responsibilities, and I wonder if it would make sense to split this into two (one for validation, one for extracting payloads). @vtereso @ncskier @EliZucker @dibyom do you know if there are any existing issues where this is being discussed? if not I'm happy to make one :D

Reply:
Aside from the [renaming issue](https://github.com/tektoncd/triggers/issues/62) and the [reusability issue](https://github.com/tektoncd/triggers/issues/55) (this issue), I don't believe there are. _Maybe_ they could be divided, but since the `TriggerBinding` is the sole place/resource that the event/payload is managed, I think it makes sense that it does both. This PR gives `TriggerBindings` separate what was previously params into input params so that matching can be done in a reusable way and output params, which are to be used for the `TriggerTemplate`. Alternatively, we could provide access to the headers emit two kinds of parameters from the `TriggerBinding` so that one could be used for validation and one for the `TriggerTemplate`. More or less, wherever the static values are colocated will not be a reusable component and since `EventListeners` are specific instantiations, I believe that was the premise behind this architecture (passing in the values to the `TriggerBinding`).

Reply:
Spoke with @ncskier, he made a point that this new conditional type could also have access to the event, where we would only need one params (the output params) in the `TriggerBinding`.

Reply:
I agree with @bobcatfish that it might be helpful to break the conditionals & header matching out of the EventBinding. I think it would be helpful to have a new issue for this.

I'll post my idea as a comment in the new issue, but for example, we could create a `TriggerCondition` resource which evaluates to `true`/`false`, and which determines whether or not the `TriggerBinding` + `TriggerTemplate` is executed/instantiated. This `TriggerCondition` could be a top-level resource with `TriggerTemplate` and `TriggerBinding` where all of these could be specified in the `EventListener`.

Reply:
Okay I made https://github.com/tektoncd/triggers/issues/71 with my thoughts! I'd like to see the `TriggerCondition` option too @ncskier :D

Reply:
Resolving this since it branched off into its own issue.